atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -279,6 +279,14 @@ private void checkLeaderAndTimelockBlocks() {
     private void checkLockAndTimestampBlocks() {
         Preconditions.checkState(lock().isPresent() == timestamp().isPresent(),
                 "Lock and timestamp server blocks must either both be present or both be absent.");
+        checkServersListHasAtLeastOneServerIfPresent(lock());
+        checkServersListHasAtLeastOneServerIfPresent(timestamp());
+    }
+
+    private static void checkServersListHasAtLeastOneServerIfPresent(Optional<ServerListConfig> serverListOptional) {
+        serverListOptional.ifPresent(
```

Comment:
Is this for Timelock or separate Timestamps and Lock blocks?

Have you verified this condition is actually true all of the time?  Is it guaranteed by the config source?

Might be better to code defensively around all of the various ways the config could indicate no servers.

Reply:
This is only applied for the Timestamp and Lock blocks.

The leader config does its own verification. The state of the world is thus:
leader with 0 --> fail in configuration
lock/timestamp with 0 --> fail here
timelock with 0 --> pass

I'd prefer to keep leader with the 1 minimum as it is, as LeaderConfig is used internally. Though as a defensive practice we can also check here.

So we're targeting:
leader with 0 --> fail in configuration **and here**
lock/timestamp with 0 --> fail here
timelock with 0 --> pass